src/main/java/com/google/api/codegen/discovery/ViewModelProvider.java

```diff
@@ -72,15 +66,28 @@ private ViewModelProvider(
     // DiscoveryProvider to generate a single SampleConfig and provide one
     // method at a time.
     SampleConfig sampleConfig =
-        new ApiaryConfigToSampleConfigConverter(methods, apiaryConfig, typeNameGenerator).convert();
-    sampleConfig = override(sampleConfig, sampleConfigOverrides);
-    ViewModel surfaceDoc = methodToViewTransformer.transform(method, sampleConfig);
-    Doc doc = snippetSetRunner.generate(surfaceDoc);
+        new ApiaryConfigToSampleConfigConverter(methods(), apiaryConfig(), typeNameGenerator())
+            .convert();
+    ObjectMapper mapper = new ObjectMapper().registerModule(new GuavaModule());
+    JsonNode tree = mapper.valueToTree(sampleConfig);
+    tree = override(tree, sampleConfigOverrides());
+
+    if (sampleOptions().noAuth()) {
```

Comment:
Seems like it might be more elegant---and readable---here to keep the surrounding plumbing in the `override` function and use it again with the `noAuth` flag formatted as an additional override.